Gantt chart for running dag doesn't show up in Airflow 1.10.11

**Apache Airflow version**: 1.10.11

**Kernel** (e.g. `uname -a`): Linux 20cb927bb9ef 5.3.0-1017-aws #18~18.04.1-Ubuntu SMP Wed Apr 8 15:12:16 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**What happened**:
Gantt chart for running dags doesn't show up in RBAC UI & gives javascript errors:
```javascript
d3.min.js:1 Error: <rect> attribute transform: Expected number, "translate(NaN,8)".
```
![Screenshot from 2020-07-14 11-31-15](https://user-images.githubusercontent.com/1238597/87410180-3dd83f00-c5c6-11ea-87f3-34b18447e81d.png)


**How to reproduce it**:
By checking the gantt chart of a **running** dag

--- a/airflow/www/views.py
+++ b/airflow/www/views.py
@@ -2493,38 +2493,33 @@ class Airflow(AirflowBaseView):  # noqa: D101  pylint: disable=too-many-public-m
             )
         )
 
-        # determine bars to show in the gantt chart
-        gantt_bar_items = []
-
         tasks = []
         for ti in tis:
-            end_date = ti.end_date or timezone.utcnow()
             # prev_attempted_tries will reflect the currently running try_number
             # or the try_number of the last complete run
             # https://issues.apache.org/jira/browse/AIRFLOW-2143
-            try_count = ti.prev_attempted_tries
-            gantt_bar_items.append((ti.task_id, ti.start_date, end_date, ti.state, try_count))
+            try_count = ti.prev_attempted_tries if ti.prev_attempted_tries != 0 else ti.try_number
             task_dict = alchemy_to_dict(ti)
+            task_dict['end_date'] = task_dict['end_date'] or timezone.utcnow()
             task_dict['extraLinks'] = dag.get_task(ti.task_id).extra_links
+            task_dict['try_number'] = try_count
             tasks.append(task_dict)
 
         tf_count = 0
         try_count = 1
         prev_task_id = ""
         for failed_task_instance in ti_fails:
-            end_date = failed_task_instance.end_date or timezone.utcnow()
-            start_date = failed_task_instance.start_date or end_date
             if tf_count != 0 and failed_task_instance.task_id == prev_task_id:
                 try_count += 1
             else:
                 try_count = 1
             prev_task_id = failed_task_instance.task_id
-            gantt_bar_items.append(
-                (failed_task_instance.task_id, start_date, end_date, State.FAILED, try_count)
-            )
             tf_count += 1
             task = dag.get_task(failed_task_instance.task_id)
             task_dict = alchemy_to_dict(failed_task_instance)
+            end_date = task_dict['end_date'] or timezone.utcnow()
+            task_dict['end_date'] = end_date
+            task_dict['start_date'] = task_dict['start_date'] or end_date
             task_dict['state'] = State.FAILED
             task_dict['operator'] = task.task_type
             task_dict['try_number'] = try_count

